test/infrastructure/docker/api/v1beta1/dockercluster_types.go

```diff
@@ -88,10 +88,13 @@ type DockerClusterStatus struct {
 // APIEndpoint represents a reachable Kubernetes API endpoint.
 type APIEndpoint struct {
 	// Host is the hostname on which the API server is serving.
-	Host string `json:"host"`
+	// +optional
+	Host string `json:"host,omitempty"`
 
 	// Port is the port on which the API server is serving.
-	Port int `json:"port"`
+	// +optional
+	// +kubebuilder:default=6443
```

Comment:
I know this is functionally equivalent but we are trying to keep all the defaulting rules written in a single point (the webhook) for sake of maintainability/testability. Could you kindly move this as well?

Reply:
Should I change the port field to a pointer so we can check if it was specified? if not, is`0` value never valid for CAPD?
The reason why I didn't put the defaulting in webhook is to avoid writing manual conversions if the field becomes a pointer.

Reply:
I think for CAPD for now it's fine to default to 6443 if the "incoming" port is 0.

Ideally we want to change this to a pointer with v1beta2